mempool/errors.go

```diff
@@ -0,0 +1,95 @@
+package mempool
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ErrTxInCache is returned to the client if we saw tx earlier
+var ErrTxInCache = errors.New("tx already exists in cache")
+
+// ErrTxTooLarge defines an error when a transaction is too big to be sent in a
+// message to other peers.
+type ErrTxTooLarge struct {
+	Max    int
+	Actual int
+}
+
+func (e ErrTxTooLarge) Error() string {
+	return fmt.Sprintf("Tx too large. Max size is %d, but got %d", e.Max, e.Actual)
+}
+
+// ErrMempoolIsFull defines an error where CometBFT and the application cannot
+// handle that much load.
+type ErrMempoolIsFull struct {
+	NumTxs      int
+	MaxTxs      int
+	TxsBytes    int64
+	MaxTxsBytes int64
+}
+
+func (e ErrMempoolIsFull) Error() string {
+	return fmt.Sprintf(
+		"mempool is full: number of txs %d (max: %d), total txs bytes %d (max: %d)",
+		e.NumTxs,
+		e.MaxTxs,
+		e.TxsBytes,
+		e.MaxTxsBytes,
+	)
+}
+
+// ErrPreCheck defines an error where a transaction fails a pre-check.
+type ErrPreCheck struct {
+	Err error
+}
+
+func (e ErrPreCheck) Error() string {
+	return e.Err.Error()
+}
+
+func (e ErrPreCheck) Unwrap() error {
+	return e.Err
+}
+
+// IsPreCheckError returns true if err is due to pre check failure.
+func IsPreCheckError(err error) bool {
+	return errors.As(err, &ErrPreCheck{})
+}
+
+var ErrTxNotFound = errors.New("transaction not found in mempool")
```

Comment:
nit: move this on top of the file so that all variables are in the same place

Reply:
@pysel thanks for your review, addressed.